architecture.tex

```diff
@@ -264,12 +268,12 @@ \subsection{Extensibility}
 }
 
 \begin{lstlisting}[caption={A minimal implementation of \texttt{sympy.gamma}.},label=fig:gamma-example]
-from sympy import Integer, Function, floor, factorial, polygamma
+from sympy import Function, floor, factorial, polygamma
 
 class gamma(Function)
     @classmethod
     def eval(cls, arg):
-        if isinstance(arg, Integer) and arg.is_positive:
+        if arg.is_integer and arg.is_positive:
```

Comment:
I would leave this as it was. `gamma` in SymPy evaluates on Integer, not integer, and regardless, evaluating based on assumptions not something I particularly want to encourage.